src/common.jl

```diff
@@ -1,32 +1,32 @@
 ######################################
 #### common naive Bayes functions ####
 ######################################
-
 """
-    to_matrix(V::Vector{Vector}}) -> M::Matrix
+    to_matrix(D::Dict{Symbol, Vector}}) -> M::Matrix
 
-convert a vector of vector into a matrix
+convert a dictionary of vectors into a matrix
 """
-function to_matrix{T <: Number}(V::Vector{Vector{T}})
-    n_lines = length(V)
-    n_lines < 1  && throw(ArgumentError("Empty input"))
-    X = zeros(n_lines, length(V[1]))
-    for i=1:n_lines
-        X[i, :] = V[i]
+function to_matrix{T <: Number, N}(V::Dict{N, Vector{T}})
+    n_features = length(V)
+    n_features < 1  && throw(ArgumentError("Empty input"))
+    X = zeros(n_features, length(V[collect(keys(V))[1]]))
+    for (i, f) in enumerate(values(V))
+        X[i, :] = f
     end
     return X
 end
 
+
 """
-    restructure_matrix(M::Matrix) -> V::Vector{Vector}
+    restructure_matrix(M::Matrix) -> V::Dict{Symbol, Vector}
```

Comment:
So, do we care about backwards compatibility? This will break other peoples code, so maybe we could maintain the `restructure_matrix(M::Matrix) -> V::Vector{Vector}` and mark it as `deprecated` and internally parse it to a `Dict{Symbol, Vector}` somehow ?